File: scripts/upload-images.js
import dotenv from 'dotenv'
import { v2 as cloudinary } from 'cloudinary'
import fs from 'fs/promises'
import path from 'path'

dotenv.config()

cloudinary.config({
  cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
  api_key: process.env.CLOUDINARY_API_KEY,
  api_secret: process.env.CLOUDINARY_API_SECRET,
  secure: true,
})

async function uploadImage(filePath, folder) {
  try {
    const result = await cloudinary.uploader.upload(filePath, {
      folder: `superhidromack/${folder}`,
      resource_type: 'image',
      use_filename: true,
      unique_filename: true,
      overwrite: false,
    })
    return { success: true, url: result.secure_url, publicId: result.public_id, file: path.basename(filePath) }
  } catch (err) {
    return { success: false, error: err.message, file: path.basename(filePath) }
  }
}

async function processDirectory(dirPath, folderName) {
  const files = await fs.readdir(dirPath, { withFileTypes: true })
  const results = []

  for (const file of files) {
    const fullPath = path.join(dirPath, file.name)
    if (file.isDirectory()) {
      const subResults = await processDirectory(fullPath, `${folderName}/${file.name}`)
      results.push(...subResults)
    } else if (/\.(png|jpe?g|webp|svg)$/i.test(file.name)) {
      console.log(`Subiendo: ${file.name}...`)
      const result = await uploadImage(fullPath, folderName)
      results.push(result)
      if (result.success) {
        console.log(`  ✅ ${result.url}`)
      } else {
        console.log(`  ❌ ${result.error}`)
      }
    }
  }

  return results
}

async function main() {
  const targetPath = process.argv[2] || '../SuperHidroMack/src/assets'
  const absolutePath = path.resolve(targetPath)

  console.log(`Escaneando: ${absolutePath}\n`)

  const results = await processDirectory(absolutePath, 'assets')

  const successful = results.filter((r) => r.success)
  const failed = results.filter((r) => !r.success)

  console.log(`\n========================================`)
  console.log(`Total: ${results.length}`)
  console.log(`Exitosas: ${successful.length}`)
  console.log(`Fallidas: ${failed.length}`)
  console.log(`========================================\n`)

  // Generar mapping JSON
  const mapping = {}
  for (const r of successful) {
    mapping[r.file] = { url: r.url, publicId: r.publicId }
  }

  const outputPath = path.resolve('image-mapping.json')
  await fs.writeFile(outputPath, JSON.stringify(mapping, null, 2))
  console.log(`Mapping guardado en: ${outputPath}`)

  if (failed.length > 0) {
    console.log('\nArchivos fallidos:')
    for (const f of failed) {
      console.log(`  - ${f.file}: ${f.error}`)
    }
  }
}

main().catch((err) => {
  console.error('Error fatal:', err)
  process.exit(1)
})

File: scripts/upload_images.py
import os
import json
import cloudinary
import cloudinary.uploader
from pathlib import Path

# Configuracion
cloudinary.config(
    cloud_name="do3kjbiy8",
    api_key="996285372526885",
    api_secret="GlgV-gGajJA15QrxxA3FnEF_ctY",
    secure=True
)

# Rutas
BASE_DIR = Path(r"C:\Users\insan\Desktop\SuperHidroMack\src\assets")
OUTPUT_JSON = Path(r"C:\Users\insan\Desktop\backend SuperHidromack\image-mapping.json")

ALLOWED_EXTENSIONS = {'.png', '.jpg', '.jpeg', '.webp', '.svg'}

def upload_file(file_path):
    """Sube un archivo a Cloudinary y retorna resultado."""
    relative = file_path.relative_to(BASE_DIR)
    folder = f"superhidromack/assets/{relative.parent.as_posix()}"
    file_name = file_path.stem
    ext = file_path.suffix.lower()

    try:
        result = cloudinary.uploader.upload(
            str(file_path),
            folder=folder,
            public_id=file_name,
            overwrite=False,
            resource_type="image" if ext != '.svg' else "raw",
            use_filename=True,
            unique_filename=False,
        )
        return {
            "success": True,
            "url": result["secure_url"],
            "public_id": result["public_id"],
            "file": relative.as_posix(),
        }
    except Exception as e:
        return {
            "success": False,
            "error": str(e),
            "file": relative.as_posix(),
        }

def main():
    results = []
    mapping = {}

    print("Escaneando: {}".format(BASE_DIR))
    print("")

    # Recorrer recursivamente
    for file_path in BASE_DIR.rglob("*"):
        if file_path.is_file() and file_path.suffix.lower() in ALLOWED_EXTENSIONS:
            print("Subiendo: {}...".format(file_path.name))
            result = upload_file(file_path)
            results.append(result)
            if result["success"]:
                print("  OK: {}".format(result['url']))
                mapping[result["file"]] = {
                    "url": result["url"],
                    "public_id": result["public_id"],
                }
            else:
                print("  ERROR: {}".format(result['error']))

    # Guardar JSON
    with open(OUTPUT_JSON, "w", encoding="utf-8") as f:
        json.dump(mapping, f, indent=2, ensure_ascii=False)

    successful = [r for r in results if r["success"]]
    failed = [r for r in results if not r["success"]]

    print("")
    print("=" * 40)
    print("Total: {}".format(len(results)))
    print("Exitosas: {}".format(len(successful)))
    print("Fallidas: {}".format(len(failed)))
    print("=" * 40)
    print("")
    print("Mapping guardado en: {}".format(OUTPUT_JSON))

    if failed:
        print("")
        print("Archivos fallidos:")
        for f in failed:
            print("  - {}: {}".format(f['file'], f['error']))

if __name__ == "__main__":
    main()
